lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -1,25 +1,67 @@
-require "pry"
+require "active_support/inflector"
 
 module RuboCop
   module Cop
     module RootCops
       class FactoryFileName < Cop
-        FILE_NAME_ERROR = "Factory name should match file name".freeze
+        # Whitelisted words feel like invariant nouns but they are really not
+        # Ex: the singular of data is datum, but we should be able to name a factory '*_data' without error
+        WHITELIST = ["data".freeze, "metadata".freeze].freeze
 
         def investigate(processed_source)
           file_path = processed_source.buffer.name
           @base_file_name = File.basename(file_path, ".rb")
+          @base_file_name_body = _generate_body(@base_file_name)
+          @base_file_name_last_word = @base_file_name.split("_").last
         end
 
         def on_send(node)
-          method_name = node.method_name
+          return if _file_name_has_error?
+
+          receiver, method_name = *node
+          _, receiver_name = *receiver
+
+          if receiver_name == :FactoryBot && method_name == :define
+            unless _file_name_pluralized?
+              add_offense(
+                node,
+                :location => :expression,
+                :severity => :fatal,
+                :message => "Factory file name should be plural (#{@base_file_name.pluralize})."
```

Comment:
IMO this almost feels like it should be a separate cop rather than having an "everything factorybot" cop. (See how the private methods cops are set up inside a namespace, maybe)

Reply:
I think the name of this cop kinda reinforces that, too: I wouldn't expect a `FactoryFileName` cop to complain about the name of my factories

Reply:
> I think the name of this cop kinda reinforces that, too: I wouldn't expect a `FactoryFileName` cop to complain about the name of my factories

I completely agree that these should be separate cops, and the private method cops are a great reference, thank you. I think one issue with splitting them up is that the file name has to be correct before checking the factory names. I will see if the `offenses` array RuboCop provides is per file or per cop. If it's per file then splitting up the cops should be no problem, but if it's per cop then I don't know how the `FactoryName` cop would know if the `FactoryFileName` cop complained or not.

Reply:
> the file name has to be correct before checking the factory names

Oh I see. That's alright then. If that's the case, don't we need to early return when we add that offense then?

Reply:
> Oh I see. That's alright then. If that's the case, don't we need to early return when we add that offense then?

[This check](https://github.com/Root-App/root-ruby-style/blob/2d322fbc18e4e02297e1cd9ca0055b538f7a89c0/lib/rubocop/cop/root_cops/factory_file_name.rb#L19) is meant to skip the entire inside of a file if the file name is incorrect (an incorrect file name is registered as a `fatal` error). It seems like there should be an easier way to early return though, so I'm going to look into it a little more.

Reply:
Maybe we could pull the early return up to `on_block` instead of `on_send` since that would result in fewer checks.

Reply:
I will give this a go and will probably have some more questions down the line. Thanks for the suggestions!

Reply:
The cops have now been split up and there's a `Helpers::Factories` that consolidates a few of the shared methods.